### PR TITLE
Fix panic for concurrent DeleteQueue & ReceiveMessage

### PR DESCRIPTION
**Pull request by sebcante**
Hi @p4tin,

Hope we can merge the below bug fix, thanks!

Fixes panic at  goaws/app/gosqs/gosqs.go:376. use case `DeleteQueue()` while something is still blocked by `ReceiveMessage()`

Fixes the below panic in `ReceiveMessage`
```[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x12bf1a0]

goroutine 50 [running]:
github.com/sebcante/goaws/app/gosqs.ReceiveMessage(0x13fe060, 0xc000088ac0, 0xc0001ce800)
        github.com/sebcante/goaws/app/gosqs/gosqs.go:376 +0x410
net/http.HandlerFunc.ServeHTTP(...)
        /Users/seb/ap/bin/gobin/go/src/net/http/server.go:1995
github.com/sebcante/goaws/app/gosqs.TestReceiveMessage_WithConcurrentDeleteQueue.func1(0xc000018b20, 0xc000163c00)
        github.com/sebcante/goaws/app/gosqs/gosqs_test.go:1081 +0x344
created by github.com/sebcante/goaws/app/gosqs.TestReceiveMessage_WithConcurrentDeleteQueue
        github.com/sebcante/goaws/app/gosqs/gosqs_test.go:1065 +0x482"
```

### CODE REVIEW

**Comment by sebcante** [comment]
Hi @p4tin, would you have any pointer on the travis error?, is that something you saw before?  happy to jump on it asap

```The command "eval go get -v -t ./... " failed. Retrying, 2 of 3.
package math/bits: unrecognized import path "math/bits" (import path does not begin with hostname)```
https://travis-ci.org/p4tin/goaws/jobs/589937679

File: app/gosqs/gosqs.go [hunk 1]
```diff
@@ -373,9 +373,17 @@ func ReceiveMessage(w http.ResponseWriter, req *http.Request) {
...
 		app.SyncQueues.RLock()
-		found := len(app.SyncQueues.Queues[queueName].Messages)-numberOfHiddenMessagesInQueue(*app.SyncQueues.Queues[queueName]) != 0
+		_, queueFound := app.SyncQueues.Queues[queueName]
+		var messageFound bool
+		if queueFound {
+			messageFound = len(app.SyncQueues.Queues[queueName].Messages)-numberOfHiddenMessagesInQueue(*app.SyncQueues.Queues[queueName]) != 0
+		}
 		app.SyncQueues.RUnlock()
-		if !found {
+		if !queueFound {
```
Comment (p4tin) [inline app/gosqs/gosqs.go]: Move this above the `var messageFound bool` so to exit earlier on error.

**Comment by p4tin** [comment]
I have never seen that error in travis CI

**Comment by p4tin** [comment]
I see it in other PRs as well - I cannot troubleshoot that right now but I will this weekend - in the meantime let me know if you find a reason why...

File: app/gosqs/gosqs.go (hunk 1 above)
Comment (sebcante) [reply app/gosqs/gosqs.go]: go idea moved it up to exit earlier, thanks!

**Pull request merged**